src/game/path.service.js

```diff
@@ -58,28 +58,39 @@ angular.module('ngGo.Game.Path.Service', [
       return;
     }
 
+    //Decrement move
+    this.move--;
+
     //Delete path choice
-    if (this.path[this.move]) {
+    if (typeof this.path[this.move] !== 'undefined') {
+      var branch = this.path[this.move];
       delete this.path[this.move];
-      this.branches--;
+      if (branch > 0) {
+        this.branches--;
+      }
     }
 
-    //Decrement move
-    this.move--;
     return this;
```

Comment:
Oh, I got it. I've changed the code to not to record the index for main branch. The change is meant to fix the code `this.path[this.move] = 1;` (it should be `i` instead).